### PR TITLE
Player/Services port from Owl

### PR DESCRIPTION
**Pull request by mbst-jamie**

### CODE REVIEW

File: atlas-api/src/main/java/org/atlasapi/output/annotation/LocationsAnnotation.java [hunk 1]
```diff
@@ -55,7 +66,14 @@ public void write(EncodedLocation entity, FieldWriter writer, OutputContext ctxt
...
             writer.writeList("available_countries", "country", policy.getAvailableCountries(), ctxt);
-            writer.writeField("service", policy.getService());
+            if (policy.getServiceRef() != null) {
+
+            }
+            if (policy.getPlayerRef() != null) {
+
+            }
+            writer.writeField("service", policy.getServiceRef());
+            writer.writeField("player", policy.getPlayerRef());
```
Comment (wwajerowicz) [inline atlas-api/src/main/java/org/atlasapi/output/annotation/LocationsAnnotation.java]: Shouldn't these two lines be moved inside the if statements?

Comment (mbst-jamie) [inline atlas-api/src/main/java/org/atlasapi/output/annotation/LocationsAnnotation.java]: Close, they shouldn't exist anymore, just hadn't finished working there yet ;p

File: atlas-api/src/main/java/org/atlasapi/output/annotation/LocationsAnnotation.java [hunk 2]
```diff
@@ -55,7 +82,21 @@ public void write(EncodedLocation entity, FieldWriter writer, OutputContext ctxt
...
             writer.writeField("availability_end", policy.getAvailabilityEnd());
             writer.writeList("available_countries", "country", policy.getAvailableCountries(), ctxt);
-            writer.writeField("service", policy.getService());
+            if (policy.getServiceRef() != null) {
+                Optional<org.atlasapi.media.entity.Service> maybeService =
+                        serviceResolver.serviceFor(policy.getServiceRef().getId().longValue());
+                if (maybeService.isPresent()) {
+                    Service service = serviceTransformer.apply(maybeService.get());
+                    writer.writeObject(serviceWriter, service, ctxt);
+                }
```
Comment (wwajerowicz) [inline atlas-api/src/main/java/org/atlasapi/output/annotation/LocationsAnnotation.java]: If a Service which doesn't exist is reffered to, I think we should log it, not just ignore it. It's a data integrity issue,

File: atlas-api/src/main/java/org/atlasapi/output/annotation/LocationsAnnotation.java [hunk 3]
```diff
@@ -55,7 +82,21 @@ public void write(EncodedLocation entity, FieldWriter writer, OutputContext ctxt
...
+            }
+            if (policy.getPlayerRef() != null) {
+                Optional<org.atlasapi.media.entity.Player> maybePlayer = playerResolver.playerFor(policy.getPlayerRef().getId().longValue());
+                if (maybePlayer.isPresent()) {
+                    Player player = playerTransformer.apply(maybePlayer.get());
+                    writer.writeObject(playerWriter, player, ctxt);
```
Comment (wwajerowicz) [inline atlas-api/src/main/java/org/atlasapi/output/annotation/LocationsAnnotation.java]: Same with Player

**Comment by wwajerowicz** [comment]
Other than two comments on the code, :+1:

**Pull request merged**